net/unbound/Makefile

```diff
@@ -23,49 +23,101 @@ PKG_CPE_ID:=cpe:/a:nlnetlabs:unbound
 PKG_BUILD_PARALLEL:=1
 PKG_FIXUP:=autoreconf
 PKG_INSTALL:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
 
 include $(INCLUDE_DIR)/package.mk
 
 define Package/unbound/Default
-  TITLE:=Validating Recursive DNS Server
-  URL:=https://www.unbound.net/
-  DEPENDS:=+libopenssl +@OPENSSL_WITH_EC
-endef
-
-define Package/unbound
-  $(call Package/unbound/Default)
   SECTION:=net
   CATEGORY:=Network
   SUBMENU:=IP Addresses and Names
   USERID:=unbound:unbound
-  TITLE+= (daemon)
+  TITLE:=Recursive DNS Server
+  URL:=https://nlnetlabs.nl/projects/unbound/about
+  DEPENDS:=+libopenssl +@OPENSSL_WITH_EC
+endef
+
+define Package/unbound-daemon
+  $(call Package/unbound/Default)
+  TITLE+= (daemon, light duty)
   DEPENDS+= +libunbound
+  VARIANT:=light
+endef
+
+define Package/unbound-daemon/description
+  This package contains the Unbound daemon with basic includes
+  necessary to meet the needs of UCI/LuCI configuration optoins.
 endef
 
-define Package/unbound/description
-  This package contains the Unbound daemon.
+define Package/unbound-daemon-heavy
+  $(call Package/unbound/Default)
+  TITLE+= (daemon, heavy duty)
+  DEPENDS+= +libunbound-heavy +libpthread +libevent2 +libevent2-pthreads
+  VARIANT:=heavy
+  PROVIDES:=unbound-daemon
+endef
+
+define Package/unbound-daemon-heavy/description
+  This package contains the Unbound daemon including 'libevent' and
+  'libpthread' to better handle large networks with heavy query loads.
+  (https://nlnetlabs.nl/documentation/unbound/howto-optimise/)
+endef
+
+define Package/libunbound
+  $(call Package/unbound/Default)
+  SECTION:=libs
+  CATEGORY:=Libraries
+  SUBMENU:=Networking
+  TITLE+= (library, light duty)
+  VARIANT:=light
+endef
+
+define Package/libunbound/description
+  This package contains the Unbound shared library with basic includes
+  necessary to meet the needs of UCI/LuCI configuration optoins.
+endef
+
+define Package/libunbound-heavy
+  $(call Package/unbound/Default)
+  SECTION:=libs
+  CATEGORY:=Libraries
+  SUBMENU:=Networking
+  TITLE+= (library, heavy duty)
+  DEPENDS+= +libpthread +libevent2 +libevent2-pthreads
+  VARIANT:=heavy
+  PROVIDES:=libunbound
+endef
+
+define Package/libunbound-heavy/description
+  This package contains the Unbound shared library including 'libevent' and
+  'libpthread' to better handle large networks with heavy query loads.
+  (https://nlnetlabs.nl/documentation/unbound/howto-optimise/)
 endef
 
 define Package/unbound-anchor
   $(call Package/unbound/Default)
-  SECTION:=net
-  CATEGORY:=Network
-  SUBMENU:=IP Addresses and Names
-  TITLE+= (DSKEY utility)
-  DEPENDS+= +unbound +libexpat
+  TITLE+= (root DSKEY)
+  DEPENDS+= +unbound-daemon +libexpat
 endef
```

Comment:
When `VARIANT` is specified for some packages but not others, possibly the unspecified ones will not be packaged (no .ipk files generated).  I haven't tried it yet, can you confirm?

Reply:
Yes. I have built using menuconfig in a number of combinations. If selected for install or loose package, then they are built correctly. No combination leaves dead ends.

Reply:
Okay I just wrote a simple `hello` package to verify this.  Packages with no `VARIANT` specified will default to a default-variant selected by `scripts/package-metadata.pl` (not variants of its dependencies).  There is no guarantee which variant will be the default unless one of them have `DEFAULT_VARIANT:=1` specified.

In unbound's case, I'd still like to suggest either specifying light as the default variant, or assigning `VARIANT` for every package.

Reply:
I marked `DEFAULT_VARIANT:=1` and tried a few purge and redo menuconfig and make combinations. Menuconfig may have behaved a little better. The builds worked well for me before. Anyway, let me know if this is better.